flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/TableStoreFactory.java

```diff
@@ -0,0 +1,205 @@
+        Map<String, String> enrichedOptions = context.getCatalogTable().getOptions();
+        Path path = tablePath(enrichedOptions, context.getObjectIdentifier());
+        try {
+            if (path.getFileSystem().exists(path)) {
+                path.getFileSystem().delete(path, true);
+            } else if (!ignoreIfNotExists) {
+                throw new TableException(
+                        String.format(
+                                "Failed to delete file store path. "
+                                        + "Reason: directory %s doesn't exist for table %s. "
+                                        + "Suggestion: please try `DROP TABLE IF EXISTS` ddl instead.",
+                                path, context.getObjectIdentifier().asSerializableString()));
+            }
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+        if (enableChangeTracking(enrichedOptions)) {
+            Context logStoreContext =
+                    new FactoryUtil.DefaultDynamicTableContext(
+                            context.getObjectIdentifier(),
+                            context.getCatalogTable().copy(filterLogStoreOptions(enrichedOptions)),
+                            context.getEnrichmentOptions(),
+                            context.getConfiguration(),
+                            context.getClassLoader(),
+                            context.isTemporary());
+            LogStoreTableFactory.discoverLogStoreFactory(
+                            Thread.currentThread().getContextClassLoader(),
+                            KafkaLogStoreFactory.IDENTIFIER)
+                    .onDropTable(logStoreContext, ignoreIfNotExists);
+        }
+    }
+
+    @Override
+    public Map<String, String> onCompactTable(
+            Context context, CatalogPartitionSpec catalogPartitionSpec) {
+        // TODO: next commit
+        return Collections.emptyMap();
+    }
+
+    @Override
+    public Set<ConfigOption<?>> requiredOptions() {
+        return Collections.emptySet();
+    }
+
+    @Override
+    public Set<ConfigOption<?>> optionalOptions() {
+        LogStoreTableFactory logStoreTableFactory =
+                LogStoreTableFactory.discoverLogStoreFactory(
+                        Thread.currentThread().getContextClassLoader(),
+                        KafkaLogStoreFactory.IDENTIFIER);
+        Set<ConfigOption<?>> options = new HashSet<>(logStoreTableFactory.optionalOptions());
+        options.addAll(logStoreTableFactory.requiredOptions());
+        options.add(BUCKET);
+        options.add(MANIFEST_TARGET_FILE_SIZE);
+        options.add(CHANGE_TRACKING);
+        return options;
+    }
+
+    // ~ Tools ------------------------------------------------------------------
+
+    private Map<String, String> filterLogStoreOptions(Map<String, String> enrichedOptions) {
+        Map<String, String> logStoreOptions = new HashMap<>();
+        LogStoreTableFactory logStoreTableFactory =
```

Comment:
For example, `CHANGELOG_MODE` in `LogOptions`.
Actually, in the table store level, user should use `log.changelog-mode`.
So here, we should just pick options with `log.` prefix, and remove prefix.